sycl/cmake/modules/SYCLVersionFromVCS.cmake

```diff
@@ -0,0 +1,32 @@
+# Helper utility to generate sycl/version.hpp file which contains various
+# identifiers which can be used to distinguish different builds between each
+# other:
+#
+# __LIBSYCL_TIMESTAMP - timestamp of the latest commit made into sycl/ directory
+# in YYYYMMDD format
+#
+# __SYCL_COMPILER_VERSION - date when configure step was launched in YYYYMMDD
+# format. Deprecated
+#
+# Note: it may not always be the case that CMake configuration step was re-run
```

Comment:
[`execute_process`](https://cmake.org/cmake/help/latest/command/execute_process.html) is run during configure step by definition.

I haven't yet fully explored the path of doing the same during build step. There is `add_custom_command`, but its result is a file; `configure_file` seems to be happening only during configure step as well - it is probably possible, but it will definitely be way more verbose than what we have now.